### ISSUE TITLE
Support DDEV DB imports

### ISSUE THREAD

**Issue by furioursus** [nystudio107/craft-scripts#64]
In DDEV, our database settings are set to nothing more than `db`, as the system handles all that on its own. This causes a problem, because when I run `./scripts/pull_db.sh`, I get an error with those values, and the message: ```sh
Unknown suffix 'd' used for variable 'port' (value 'db')
mysqldump: Error while setting value 'db' to 'port'```

### Describe the solution you would like
I’d like for the script to accommodate this issue when validating, as we don‘t have direct access to the ports.

**Comment by khalwat** [comment]
I doubt that the `port` setting in DDEV is set to `db` -- that's not a valid Unix port.

In any event, I'm open to PRs, but I don't use DDEV myself so...

**Comment by furioursus** [comment]
Understandable, though what I mean is that the `db` value is used in place of an ephemeral port number that changes every time it starts up, so it's just impossible for us to anticipate. Folks have been asking [as far back as 2018](https://github.com/drud/ddev/issues/941) for a more concrete way to get at that information, so I'll see if anything turns up!

**Comment by furioursus** [comment]
Adding this for posterity in case I find myself asking this question again in the future (hi, future me!) and/or someone wants to just know how to get the port after DDEV is started using `ddev describe` or `ddev describe -j`: https://github.com/drud/ddev/issues/691#issuecomment-611109429

**Comment by khalwat** [comment]
Gotcha. Well, I know ddev already provides ways to import databases, so you could use the script to pull the db down from production, and then use the ddev command to import it. Slightly less convenient though.

Are we sure ddev has no built in functionality to pull down remote dbs?

**Comment by terryupton** [comment]
@onebrightlight - did you manage to resolve this and what are your steps to getting it working to `pull_db.sh` when working with DDEV?

**Comment by furioursus** [comment]
i no longer have to run scripts usually for this as they're more often a manual process, but i believe you should be able to modify the bash script to use a command along the lines of `ddev mysql` per https://ddev.readthedocs.io/en/latest/users/usage/database-management/

i'll give it a look this afternoon and see if i canny cobble something together because i can see how customizing some of @khalwat's scripts for a DDEV-specific setup would come in handy down the road for me, too.

**Comment by furioursus** [comment]
@terryupton now that i’m looking at it, i think the lowest effort solution here is going to be setting your DDEV instance’s port with [`host_db_port`](https://ddev.readthedocs.io/en/latest/users/configuration/config/#host_db_port) per [their guide on using DDEV with database GUIs](https://ddev.readthedocs.io/en/latest/users/usage/database-management/#database-guis).

It’s still not 100% ideal, but i think that this will get you going 99% of the time.

**Comment by terryupton** [comment]
Thanks @onebrightlight - this is actually what I tried and was successful on a project on Monday. Tried the exact same thing yesterday on another project I converted to DDEV and it just wouldn't connect/work, so thought I had maybe gotten lucky on the first one.  Have tried this morning with another new converted Craft site and it has worked successfully again, connecting over `127.0.0.01` and the `DDEV describe` db port. 

I will re-look at that failed project again and might try explicitly setting the DB port for it. 

Thanks for your input.
